src/background_threads/autoav.ts

```diff
@@ -0,0 +1,124 @@
+import { HubConnectionBuilder } from '@microsoft/signalr';
+import nodeFetch from 'node-fetch';
+import { parentPort, MessagePort } from 'node:worker_threads';
+import VmixRecordingService from '../services/VmixRecordingService';
+import FMSMatchStatus from '../models/FMSMatchState';
+import attemptRename from '../utils/recording';
+
+// Parent Port
+let parent = parentPort!;
+
+// Last Match "Start"
+let lastMatchStartTime: Date | null;
+let lastMatchStartData: FMSMatchStatus | null;
+let lastState: FMSMatchStatus | null;
+
+const logToParent = (msg: string) => {
+    parent.postMessage('AutoAV: ' + msg);
+}
+
+if (parent === null) {
+    parent = {
+        postMessage: console.log,
+    } as any as MessagePort;
+}
+
+// Notify Parent logs that we're running
+logToParent('AutoAV Service Started');
+
+// Create new VMix Service
+const vmixService = new VmixRecordingService({
+    baseUrl: 'http://127.0.0.1:8000/api',
+    username: 'user',
+    password: 'pass',
+});
+
+// Build a connection to the SignalR Hub
+const conn = new HubConnectionBuilder()
+    .withUrl('http://10.0.100.5:8189/infrastructureHub')
```

Comment:
It'd be nice to get some of these hardcoded strings into the config file added in #3